projects/packages/waf/src/exceptions/class-waf-exception.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Base custom exception for the WAF package.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack-waf
+ */
+
+namespace Automattic\Jetpack\Waf;
+
+use Exception;
+use WP_Error;
+
+/**
+ * WAF exception.
+ */
+class Waf_Exception extends Exception {
+
+	/**
+	 * Error slug which maps to WP_Error::$code.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'waf_error';
```

Comment:
The slug _could_ be a protected constant instead (and then called with `static::` below). Not much benefit of that other than that we can be sure that it was never changed for any of the exceptions in any case imaginable. So only a very slight improvement, if you want to do it. (Due to version restrictions I think we can not actually make the constant protected, but still, even if it is just a public one it would have the same benefit).

Reply:
https://github.com/Automattic/jetpack/pull/29108/commits/5db57fefbc9dacb7e3474d1a9b622f76a04f2bb2 :+1: :+1: